src/dynamic_programming/zero_matrix.md

```diff
@@ -0,0 +1,86 @@
+<!--?title Finding the largest zero submatrix -->
+
+# Finding the largest zero submatrix
+
+You are given a matrix with `n` rows and `m` columns. Find the largest submatrix consisting of only zeros (a submatrix is a rectangular area of the matrix).
+
+## Algorithm
+
+Elements of the matrix will be `a[i][j]`, where `i = 0...n - 1`, `j = 0... m - 1`. For simplicity, we will consider all non-zero elements equal to 1.
+
+### Step 1: Auxiliary dynamic
+
+First, we calculate the following auxiliary matrix: `d[i][j]`, nearest row that has a 1 above `a[i][j]`. Formally speaking, `d[i][j]` is the largest row number (from `1` to `i`), in which there is a element equal to `1` in the `j`-th column. 
+While iterating from top-left to bottom-right, when we stand in row `i`, we know the values from the previous row, so, it is enough to update just the elements with value `1`. We can save the values in a simple array `d[i]`, `i = 1...m - 1`:
```

Comment:
Can you add a short sentence, why we can only use a simple array, and don't require a matrix. This is only possible because of the way we solve the second part of the problem.